### PR TITLE
Aggregation expression pushdown

### PR DESCRIPTION
**Pull request by penghuo**
## Description of changes
1. Add the aggregation push down logic. 

## limitation
1. Without logical plan optimization, only support two types of aggregation push down (1) agg-relation, (2) agg-filter-relation
2. Expression over aggregation is not supported yet, which is only required by SQL.
3. The bucket size is limited as 1000 now.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by anirudha — APPROVED** [review]

**Comment by dai-chen** [comment]
Issues related:

 1. https://github.com/opendistro-for-elasticsearch/sql/issues/276
 2. https://github.com/opendistro-for-elasticsearch/sql/issues/174
 3. https://github.com/opendistro-for-elasticsearch/sql/issues/364

File: sql/src/main/java/com/amazon/opendistroforelasticsearch/sql/sql/parser/context/QuerySpecification.java [hunk 1]
```diff
@@ -117,13 +125,22 @@ public Void visitGroupByElement(GroupByElementContext ctx) {
...
     }
+
+    /**
+     * Get original text in query.
+     */
+    private String getTextInQuery(ParserRuleContext ctx) {
+      Token start = ctx.getStart();
+      Token stop = ctx.getStop();
+      return queryString.substring(start.getStartIndex(), stop.getStopIndex() + 1);
+    }
```
Comment (dai-chen) [inline sql/src/main/java/com/amazon/opendistroforelasticsearch/sql/sql/parser/context/QuerySpecification.java:143]: np: same as `AstBuilder.getTextInQuery()` and can move to some parsing util class?

**Review by dai-chen — APPROVED** [review]
LGTM. Thanks for the changes!

File: sql/src/main/java/com/amazon/opendistroforelasticsearch/sql/sql/parser/context/QuerySpecification.java (hunk 1 above)
Comment (penghuo) [reply sql/src/main/java/com/amazon/opendistroforelasticsearch/sql/sql/parser/context/QuerySpecification.java]: done.

**Pull request merged**